R/gtrends.R

```diff
@@ -163,14 +170,51 @@ gtrends <- function(query, geo, cat, ch, ...) {
 
 #' @rdname gtrends
 #' @export
-gtrends.default <- function(query, geo, cat, ch, ...) {
+gtrends.default <- function(query, 
+                            geo, 
+                            cat, 
+                            ch, 
+                            res = "week",
+                            start_date = as.Date("2004-01-01"),
+                            end_date = as.Date(Sys.time()),
+                            ...){
 
-  if (missing(geo)) geo <- "all"
+  if (missing(geo)) geo <- ""
   if (missing(cat)) cat <- "0"
   if (missing(ch))  ch  <- .getDefaultConnection()
 
   stopifnot(is.character(query),
-            is.vector(query))
+            is.vector(query),
+            all(res %in% c("week", "day")),
+            length(res) == 1)
+  
+  ## Verify the date
+  regex <- "^(19|20)\\d\\d[- /.](0[1-9]|1[012])[- /.](0[1-9]|[12][0-9]|3[01])$"
+
+  if(!grepl(regex, start_date)){
```

Comment:
(Improv)

```r
dt <- strptime(datestring, "%Y-%m-%d")
if (any(is.na(dt)) stop("Some nasty message here.")
```

I do that all the time. I have a 'date guessing' draft package which rotates over different formats til it finds one (ie 2015-11-24, 2015/11/24, 20151124, 24 Nov 2015, ...) but that never got finished.  THAT would help.

Strictly no need for lubridate or whatnot here. Base R is dandy.